xlml/utils/gpu.py

```diff
@@ -304,48 +124,163 @@ def create_resource(
+    metadata = create_metadata({
+        # "install-nvidia-driver": "True",
+        "install-nvidia-driver": "False",
+        "proxy-mode": "project_editors",
+        "ssh-keys": f"cloud-ml-auto-solutions:{ssh_keys.public}",
+    })
+
+    accelerators = [
+        compute_v1.AcceleratorConfig(
+            accelerator_count=accelerator.count,
+            accelerator_type=f"projects/{gcp.project_name}/zones/{gcp.zone}/acceleratorTypes/{accelerator.accelerator_type}",
+        )
+    ]
+    service_account = compute_v1.ServiceAccount(
+        # email = "cloud-auto-ml-solutions@google.com",
+        scopes=["https://www.googleapis.com/auth/cloud-platform"]
+    )
+
+    instance_client = compute_v1.InstancesClient()
+    network_link = "global/networks/default"
+    # Use the network interface provided in the network_link argument.
+    network_interface = compute_v1.NetworkInterface()
+    network_interface.network = network_link
+
+    if external_access:
+      access = compute_v1.AccessConfig()
+      access.type_ = compute_v1.AccessConfig.Type.ONE_TO_ONE_NAT.name
+      access.name = "External NAT"
+      access.network_tier = access.NetworkTier.PREMIUM.name
+      network_interface.access_configs = [access]
+
+    # Collect information into the Instance object.
+    instance = compute_v1.Instance()
+    instance.network_interfaces = [network_interface]
+    instance.name = instance_name
+    instance.disks = disks
+    if re.match(r"^zones/[a-z\d\-]+/machineTypes/[a-z\d\-]+$", machine_type):
+      instance.machine_type = machine_type
+    else:
+      instance.machine_type = f"zones/{zone}/machineTypes/{machine_type}"
+
+    instance.scheduling = compute_v1.Scheduling()
+    if accelerators:
+      instance.guest_accelerators = accelerators
+      instance.scheduling.on_host_maintenance = (
+          compute_v1.Scheduling.OnHostMaintenance.TERMINATE.name
+      )
+
+    if metadata:
+      instance.metadata = metadata
+
+    if service_account:
+      instance.service_accounts = [service_account]
+
+    if spot:
+      # Set the Spot VM setting
+      instance.scheduling.provisioning_model = (
+          compute_v1.Scheduling.ProvisioningModel.SPOT.name
+      )
+      instance.scheduling.instance_termination_action = instance_termination_action
+
+    if delete_protection:
+      # Set the delete protection bit
+      instance.deletion_protection = True
+
+    # Prepare the request to insert an instance.
+    request = compute_v1.InsertInstanceRequest()
+    request.zone = zone
+    request.project = project_id
+    request.instance_resource = instance
+
+    # Wait for the create operation to complete.
+    logging.info(f"Creating the {instance_name} instance in {zone}...")
+
+    operation = instance_client.insert(request=request)
+    return operation.name
+
+  @task.sensor(poke_interval=60, timeout=1800, mode="reschedule")
+  def wait_for_resource_creation(operation_name: airflow.XComArg):
```

Comment:
For accelerator provision, can we get timeout from task, something like https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/apis/task.py#L59 in TPU? So users can define it.

Reply:
Added to be consistent.